File: Desafio_Rivia/5_.js
const express = require('express');
const { MongoClient } = require('mongodb');

const app = express();
app.use(express.json()); // Para interpretar JSON no corpo da requisição

// Conectar ao MongoDB
const uri = 'mongodb://localhost:27017/';
const client = new MongoClient(uri);

async function conectarMongo() {
    await client.connect();
    console.log('Conectado ao MongoDB');
}

const db = client.db('meu_banco');
const colecao = db.collection('minha_colecao');

// Rota para inserir dados
app.post('/inserir', async (req, res) => {
    const dados = req.body;
    try {
        const resultado = await colecao.insertOne(dados);
        res.json({
            status: 'Dados inseridos com sucesso',
            id: resultado.insertedId
        });
    } catch (error) {
        console.error('Erro ao inserir dados:', error);
        res.status(500).json({ status: 'Erro ao inserir dados' });
    }
});

// Iniciar o servidor e conectar ao MongoDB
app.listen(5000, () => {
    console.log('Servidor rodando na porta 5000');
    conectarMongo().catch(console.error);
});

File: Desafio_Rivia/5_.py
from flask import Flask, request, jsonify
from pymongo import MongoClient

app = Flask(__name__)
client = MongoClient('mongodb://localhost:27017/')
db = client['meu_banco']
colecao = db['minha_colecao']

@app.route('/inserir', methods=['POST'])
def inserir_dados():
    dados = request.json
    resultado = colecao.insert_one(dados)
    return jsonify({"status": "Dados inseridos com sucesso", "id": str(resultado.inserted_id)})

if __name__ == '__main__':
    app.run(debug=True)

# Executar o código e acessar http://localhost:5000/inserir
